src/scala/microsoft-spark-2-3/src/main/scala/org/apache/spark/util/dotnet/Utils.scala

```diff
@@ -191,6 +193,45 @@ object Utils extends Logging {
       .mkString(".")
   }
 
+  /**
+   * Validates the normalized spark version by verifying:
+   *   - Spark version starts with sparkMajorMinorVersionPrefix.
+   *   - If ignoreSparkPatchVersionCheck spark conf is
+   *     - true: valid
+   *     - false: check if the spark version is in supportedSparkVersions.
+   * @param conf Spark conf.
+   * @param sparkVersion The spark version.
+   * @param normalizedSparkVersion: The normalized spark version.
+   * @param supportedSparkMajorMinorVersionPrefix The spark major and minor version to validate against.
+   * @param supportedSparkVersions The set of supported spark versions.
+   */
+  def validateSparkVersions(
+      conf: SparkConf,
+      sparkVersion: String,
+      normalizedSparkVersion: String,
+      supportedSparkMajorMinorVersionPrefix: String,
+      supportedSparkVersions: Set[String]): Unit = {
+    val ignorePatchVersion = conf.get(DOTNET_IGNORE_SPARK_PATCH_VERSION_CHECK)
+
+    if (!normalizedSparkVersion.startsWith(s"$supportedSparkMajorMinorVersionPrefix.")) {
+      throw new IllegalArgumentException(
+        s"Unsupported spark version used: $sparkVersion. " +
+          s"Normalized spark version used: $normalizedSparkVersion. " +
+          s"Supported spark major.minor version: $supportedSparkMajorMinorVersionPrefix")
```

Comment:
Let's quote the version string so that there is no confusion with dots in the version.
```suggestion
        s"Unsupported spark version used: '$sparkVersion'. " +
          s"Normalized spark version used: '$normalizedSparkVersion'. " +
          s"Supported spark major.minor version: '$supportedSparkMajorMinorVersionPrefix'")
```

Reply:
Added quotes.